lib/vtls/openssl.c

```diff
@@ -4112,6 +4112,17 @@ CURLcode Curl_ossl_ctx_init(struct ossl_ctx *octx,
   }
 
   SSL_CTX_set_options(octx->ssl_ctx, ctx_options);
+  SSL_CTX_set_read_ahead(octx->ssl_ctx, 1);
+
+  /* Max TLS1.2 record size 0x4000 + 0x800.
+     OpenSSL supports processing "jumbo TLS record" (8 TLS records) in one go
+     for some algorithms, so match that here.
+     Experimentation shows that a slightly larger buffer is needed to avoid short reads.
+
+     However using a to large buffer (8 packets) actually decreases performance.
+     4 packets is better.
+   */
+  SSL_CTX_set_default_read_buffer_len(octx->ssl_ctx, 0x401e * 4);
```

Comment:
Out of curiosity, why 0x401e? The comment above would make me think 0x4800

Reply:
`tshark` says:
```  
 Reassembled TCP length: 16413
    TLSv1.2 Record Layer: Application Data Protocol: Hypertext Transfer Protocol
        Content Type: Application Data (23)
        Version: TLS 1.2 (0x0303)
        Length: 16408
```

16413 is 0x401d. I'll try to find out whether that is more or less constant, or whether different TLS versions or settings could change that (well obviously according to the comment it can go up to +0x800, but it talks about compression, which is usually turned off due to security issues)